src/helpers/prf_compute/multiplicative_inverse.rs

```diff
@@ -0,0 +1,170 @@
+use std::cmp;
+
+use num_bigint::{BigInt, BigUint, ToBigInt};
+use num_integer::Integer;
+use num_traits::{identities::Zero, One, Signed};
+
+fn extended_binary_gcd(m: &BigUint, n: &BigUint) -> (BigInt, BigInt, BigUint) {
+    #[inline]
+    fn count_twos_multiple(x: &BigUint) -> u64 {
+        x.trailing_zeros().unwrap_or(0)
+    }
+
+    // Assume non zero
+    assert!(!m.is_zero());
+    assert!(!n.is_zero());
+
+    let mut x_ebgcd = m.clone();
+    let mut y_ebgcd = n.clone();
+    let mut g_ebgcd = BigUint::one();
+
+    // find common factors of 2
+    let shift = cmp::min(count_twos_multiple(&x_ebgcd), count_twos_multiple(&y_ebgcd));
+
+    x_ebgcd >>= shift;
+    y_ebgcd >>= shift;
+    g_ebgcd <<= shift;
+
+    let mut u_ebgcd = x_ebgcd.clone();
+    let mut v_ebgcd = y_ebgcd.clone();
+
+    let mut a_ebgcd = BigInt::one();
+    let mut b_ebgcd = BigInt::zero();
+    let mut c_ebgcd = BigInt::zero();
+    let mut d_ebgcd = BigInt::one();
```

Comment:
These variable names are really hard for me to work with. I have to keep looking back up and these re-assignments to understand what they all are.

What is "u_ebgcg"? Oh, it's a clone of x_ebgcg... what's that? Oh, it's a clone of m... what was m again? Is it the prime?

I'd love maybe a comment at least (if not better names). For example, what is the "_ebgcd" suffix adding here? (Aside from extra characters...)

Reply:
Changed the variable names to being a bit more meaningful + also added comments to the function to tell what it actually does